splunk/tests/splunk_test_prep.go

```diff
@@ -0,0 +1,65 @@
+package main
+
+import (
+	"fmt"
+	"log"
+	"os"
+	"os/exec"
+	"strings"
+	"time"
+)
+
+// BuildSplunk builds the Splunk image used for testing Splunk queries.
+func BuildSplunk() {
+	buildCmd := exec.Command("podman", "build", "-t", "test_splunk", "./splunk")
+
+	buildCmd.Stdout = os.Stdout
+	buildCmd.Stderr = os.Stderr
+	err := buildCmd.Run()
+	if err != nil {
+		log.Fatalf("Failed to build image: %v", err)
+	}
+	log.Println("Image build done successfully!")
+}
+
+// RunSplunk deploy the Splunk container used for testing Splunk queries.
+func RunSplunk() string {
+	cmd := exec.Command("podman", "run", "-d", "-e", "SPLUNK_START_ARGS=--accept-license", "-e", "SPLUNK_PASSWORD=Password", "-p", "8000:8000", "-p", "8089:8089", "localhost/test_splunk:latest")
+	output, err := cmd.CombinedOutput()
+	if err != nil {
+		log.Fatalf("Failed to start command: %v", err)
+	}
+	containerID := strings.TrimSpace(string(output))
+	return containerID
+}
+
+// VerifyServiceIsUp continuously monitors the Splunk container to
+// ensure it is up and ready for use in tests. It achieves this by repeatedly
+// checking the status endpoint of the Splunk API over a period of two minutes.
+func VerifyServiceIsUp() {
+	timeoutStart := time.Now().Unix()
+	splunkURL := "https://localhost:8089/services/server/status"
+	username := "admin"
+	password := "Password"
+
+	for {
+		cmd := exec.Command("curl", "-s", "-o", "/dev/null", "-w", "%{http_code}", "-u",
+			fmt.Sprintf("%s:%s", username, password), "--insecure", splunkURL)
+		output, _ := cmd.CombinedOutput()
```

Comment:
```suggestion
		output, err := cmd.CombinedOutput()
```
we need to handle this error.

Reply:
Do we? It'll return an error as long as it's not getting 200. meaning the log will be filled with exit codes over and over until the loop will resolve either after 2 minutes or by getting 200.